mac/Keyman4MacIM/Keyman4MacIM/KMInputMethodEventHandler.m

```diff
@@ -917,9 +685,292 @@ - (void)postKeyPressToFrontProcess:(CGKeyCode)code from:(CGEventSourceRef) sourc
         }
     }
 
-    [self.AppDelegate postKeyboardEventWithSource:source code:code postCallback:^(CGEventRef eventToPost) {
+    [self.appDelegate postKeyboardEventWithSource:source code:code postCallback:^(CGEventRef eventToPost) {
         CGEventPostToPSN((ProcessSerialNumberPtr)&psn, eventToPost);
     }];
 }
 
+//MARK: Core-related key processing
+
+- (void)checkTextApiCompliance:(id)client {
+  // if TextApiCompliance object is null or stale,
+  // then create a new one for the current application
+  
+  if ((self.apiCompliance == nil) || (![self.apiCompliance.clientApplicationId isEqualTo:self.clientApplicationId])) {
+    self.apiCompliance = [[TextApiCompliance alloc]initWithClient:client applicationId:self.clientApplicationId];
+    [self.appDelegate logDebugMessage:@"KMInputMethodHandler initWithClient checkTextApiCompliance: %@", _apiCompliance];
+  } else if (self.apiCompliance.isComplianceUncertain) {
+    // if it is valid but compliance is uncertain, then test it
+    [self.apiCompliance testCompliance:client];
+  }
+}
+
+- (void) handleContextChangedByLowLevelEvent {
+  if (self.appDelegate.contextChangedByLowLevelEvent)
+  {
```

Comment:
Thanks for the thorough re-review. I had missed a chunk of those. A few I am going to deal with when I use the core action API change and with one other cleanup. I need to address one review comment on modifier flags and then I think it will be ready.